exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java

```diff
@@ -369,24 +370,33 @@ AvaticaFactory getFactory() {
     return factory;
   }
 
+  private static void closeOrWarn(final AutoCloseable autoCloseable, final String message, final Logger logger) {
```

Comment:
Sounds good, but let's leave that until DrillAutoCloseables and AutoCloseables are merged, along with the change you suggested earlier/above.